pyuvsim/tests/test_simsetup.py

```diff
@@ -884,48 +884,91 @@ def test_uvfits_to_config():
-        pyuvsim.create_mock_catalog(time, 'invalid_catalog_name')
-
+    # For each mock catalog, verify the Ra/Dec source positions against a saved text catalog.
     radec_catalog = pyradiosky.SkyModel()
-    for arr in arrangements:
-        radec_catalog.read_text_catalog(
-            os.path.join(SIM_DATA_PATH, 'test_catalogs', text_catalogs[arr])
-        )
-        assert np.all(radec_catalog == cats[arr])
+    radec_catalog.read_text_catalog(
+        os.path.join(SIM_DATA_PATH, 'test_catalogs', text_cat)
+    )
+    assert np.all(radec_catalog == cat)
+
 
-    cat, mock_kwds = pyuvsim.simsetup.create_mock_catalog(time, 'random', save=True)
+def test_saved_mock_catalog():
+    time = Time(2458098.27471265, scale='utc', format='jd')
+    cat, mock_kwds = pyuvsim.simsetup.create_mock_catalog(time, 'random', Nsrcs=100, save=True)
     loc = eval(mock_kwds['array_location'])
     loc = EarthLocation.from_geodetic(loc[1], loc[0], loc[2])  # Lon, Lat, alt
     fname = 'mock_catalog_random.npz'
     alts_reload = np.load(fname)['alts']
     cat.update_positions(time, loc)
     alt, az = cat.alt_az
-    assert np.all(alt > np.radians(30.))
-    assert np.allclose(alts_reload, np.degrees(alt))
     os.remove(fname)
+    assert np.allclose(alts_reload, np.degrees(alt))
+
+
+@pytest.mark.parametrize('min_alt', [-20, 0, None, 50])
+def test_randsource_minalt(min_alt):
+    time = Time(2458098.27471265, scale='utc', format='jd')
+    cat, mock_kwds = pyuvsim.simsetup.create_mock_catalog(
+        time, 'random', Nsrcs=100, min_alt=min_alt
+    )
+    loc = eval(mock_kwds['array_location'])
+    loc = EarthLocation.from_geodetic(loc[1], loc[0], loc[2])  # Lon, Lat, alt
+    cat.update_positions(time, loc)
+    alt, az = cat.alt_az
+    if min_alt is None:
+        min_alt = 30    # Checking default
+    assert np.all(alt >= np.radians(min_alt))
+
+
+def test_randsource_distribution():
+    # Check that random sources are uniformly distributed per solid angle
+
+    astropy_healpix = pytest.importorskip("astropy_healpix")
+    Nsrcs = 40000
+    time = Time(2458098.27471265, scale='utc', format='jd')
+    cat, mock_kwds = pyuvsim.simsetup.create_mock_catalog(
+        time, 'random', Nsrcs=Nsrcs, min_alt=-90, rseed=2458098
+    )
+    loc = eval(mock_kwds['array_location'])
+    loc = EarthLocation.from_geodetic(loc[1], loc[0], loc[2])  # Lon, Lat, alt
+    cat.update_positions(time, loc)
+    alt, az = cat.alt_az
+
+    # Bin into low-res HEALPix pixels.
+    nside = 32
+    npix = 12 * nside**2
+    hp = astropy_healpix.HEALPix(nside)
+
+    inds = hp.lonlat_to_healpix(Longitude(az, unit='rad'), Latitude(alt, unit='rad'))
+    un, counts = np.unique(inds, return_counts=True)
+
+    # counts should be Poisson-distributed with rate lambda = nsrcs / npix
+    # variance and mean should be close to lambda
+    lam = Nsrcs / npix
+    assert np.isclose(np.mean(counts), lam, atol=1.0)
+    assert np.isclose(np.var(counts), lam, atol=1.0)
```

Comment:
This is a clever way to handle testing a random distribution. The only thing I worry about slightly with random things in tests are flaky tests that fail randomly. Do you have a sense for how likely this is to fail with these tolerances?

Reply:
The random generator is given the same seed for each run, so it should always turn up the same points. That said, I just ran it 500 times without the fixed rseed, so it's actually random, and didn't have any failures.

Reply:
Ah, I didn't catch that it was seeded. I'm good with that, thanks.